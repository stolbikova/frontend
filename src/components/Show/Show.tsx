import styled from "styled";
import * as React from "react";
import {ShowI} from '../../types';
import {CMS_URL} from "../../constants";

interface ShowProps extends ShowI {
}

const Article = styled.article`
   display: flex;
   flex-direction: column;
   margin-bottom: 10px;
`;
const Wrapper = styled.div`
  display: flex;
  justify-content: center;
`;
const Title = styled.span`
  font-size: 32px;
  line-height: 34px;
  font-weight: 700;
`;
const SubTitle = styled.span`
  font-size: 20px;
  line-height: 24px;
`;
const Image = styled.img`
  min-width: 320px;
  width: 100%;
  margin-top: 10px;
`;
export const Show: React.FC<ShowProps> = ({title, product_image_url, episodes}) => {
  return (
    <Article>
      <Wrapper>
        <Title>{title}</Title>
      </Wrapper>
      <Wrapper>
        <SubTitle>Number of episodes {episodes}</SubTitle>
      </Wrapper>
      <Image src={`${CMS_URL}${product_image_url}`} />
    </Article>
  );
};
