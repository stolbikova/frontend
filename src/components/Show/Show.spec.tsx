import * as React from "react";
import * as renderer from "react-test-renderer";
import {Show} from "./Show";
import {getCartesianProduct} from "../../testing/cartesianProduct";
import {showMock} from "../../testing/show.props.mock";

describe("renders Show component", () => {
  const propsList = getCartesianProduct(
    {
      ...showMock
    }
  );

  propsList.forEach((props) => {
    const propsString = JSON.stringify(props);
    it(`should render with "${propsString}"`,  () => {
      const component = renderer.create(<Show {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
