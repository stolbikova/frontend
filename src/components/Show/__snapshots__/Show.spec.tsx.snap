// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Show component should render with "{"episodes":1,"id":"","product_image_url":"","title":"title1"}" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 32px;
  line-height: 34px;
  font-weight: 700;
}

.c3 {
  font-size: 20px;
  line-height: 24px;
}

.c4 {
  min-width: 320px;
  width: 100%;
  margin-top: 10px;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      title1
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c3"
    >
      Number of episodes 
      1
    </span>
  </div>
  <img
    className="c4"
    src="http://localhost:4000"
  />
</article>
`;

exports[`renders Show component should render with "{"episodes":1,"id":"","product_image_url":"","title":"title2"}" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 32px;
  line-height: 34px;
  font-weight: 700;
}

.c3 {
  font-size: 20px;
  line-height: 24px;
}

.c4 {
  min-width: 320px;
  width: 100%;
  margin-top: 10px;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      title2
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c3"
    >
      Number of episodes 
      1
    </span>
  </div>
  <img
    className="c4"
    src="http://localhost:4000"
  />
</article>
`;

exports[`renders Show component should render with "{"episodes":2,"id":"","product_image_url":"","title":"title1"}" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 32px;
  line-height: 34px;
  font-weight: 700;
}

.c3 {
  font-size: 20px;
  line-height: 24px;
}

.c4 {
  min-width: 320px;
  width: 100%;
  margin-top: 10px;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      title1
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c3"
    >
      Number of episodes 
      2
    </span>
  </div>
  <img
    className="c4"
    src="http://localhost:4000"
  />
</article>
`;

exports[`renders Show component should render with "{"episodes":2,"id":"","product_image_url":"","title":"title2"}" 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  font-size: 32px;
  line-height: 34px;
  font-weight: 700;
}

.c3 {
  font-size: 20px;
  line-height: 24px;
}

.c4 {
  min-width: 320px;
  width: 100%;
  margin-top: 10px;
}

<article
  className="c0"
>
  <div
    className="c1"
  >
    <span
      className="c2"
    >
      title2
    </span>
  </div>
  <div
    className="c1"
  >
    <span
      className="c3"
    >
      Number of episodes 
      2
    </span>
  </div>
  <img
    className="c4"
    src="http://localhost:4000"
  />
</article>
`;
